### PR TITLE
[STRMCMP-616] Improvements to integration test stability

### PR DESCRIPTION
**Pull request by mwylde**
This PR includes a bunch of improvements to integration test stability and usability.

* Improves the "are all tasks running" logic
* Retries FlinkApplication updates on conflicts 
* Uses a bad image for the force rollback test instead of too much memory
* Updates test VM to bionic
* Added more log lines to help diagnose where tests get stuck
* Logging is now plaintext instead of json

### CODE REVIEW

File: integ/utils/utils.go [hunk 1]
```diff
@@ -402,7 +406,25 @@ func (f *TestUtil) FlinkAPIGet(app *flinkapp.FlinkApplication, endpoint string)
...
-func (f *TestUtil) WaitForAllTasksInState(name string, state string) error {
+func vertexRunning(vertex map[string]interface{}) bool {
+	if vertex["status"] != "RUNNING" {
+		return false
+	}
+
+	if val, ok := vertex["tasks"]; ok {
+		parallelism := int(vertex["parallelism"].(float64))
+		tasks := val.(map[string]interface{})
+		if int(tasks["RUNNING"].(float64)) != parallelism {
```
Comment (glaksh100) [inline integ/utils/utils.go]: Unfortunate that the status does not reflect this :(

File: integ/simple_test.go [hunk 2]
```diff
@@ -208,19 +195,10 @@ func (s *IntegSuite) TestSimple(c *C) {
...
-			},
-			Limits: corev1.ResourceList{
-				corev1.ResourceCPU:    resource.MustParse("0.2"),
-				corev1.ResourceMemory: resource.MustParse("200Mi"),
-			},
-		}
 		// and we should be able to roll forward by resubmitting with a fixed config
 		updateAndValidate(c, s, config.Name, func(app *v1beta1.FlinkApplication) {
-			app.Spec.TaskManagerConfig.Resources = &TaskManagerFixedResources
+			app.Spec.Image = NewImage
```
Comment (glaksh100) [inline integ/simple_test.go]: This is so much better. Apologies for causing issues with the previous version of the test. :(

File: pkg/controller/flinkapplication/flink_state_machine.go [hunk 3]
```diff
@@ -239,7 +239,7 @@ func (s *FlinkStateMachine) handleNewOrUpdating(ctx context.Context, application
 func (s *FlinkStateMachine) deployFailed(ctx context.Context, app *v1beta1.FlinkApplication) (bool, error) {
 	hash := flink.HashForApplication(app)
 	s.flinkController.LogEvent(ctx, app, corev1.EventTypeWarning, "RolledBackDeploy",
-		fmt.Sprintf("Successfull rolled back deploy %s", hash))
+		fmt.Sprintf("Successful rolled back deploy %s", hash))
```
Comment (glaksh100) [inline pkg/controller/flinkapplication/flink_state_machine.go]: Successfully?

Comment (mwylde) [reply pkg/controller/flinkapplication/flink_state_machine.go]: Thanks, the linter automatically fixed this...

**Pull request merged**